### PR TITLE
First pass at adding a Widget component to Gift Entry

### PR DESCRIPTION
**Pull request by lparrott**
- We've added a placeholder widget component for entering one to many GAU Allocations

# Critical Changes

# Changes

# Issues Closed

# New Metadata

- geFormWidget
- geFormWidgetAllocation
- geFormWidgetRow

# Deleted Metadata

# Testing Notes
- Some notes can be found in Quip: https://salesforce.quip.com/snjkAgyMh9UP

### CODE REVIEW

**Comment by annejackson** [comment]
Added @bdvorachek  and @ptewson-sfdo for code review since Vlad is out on PTO for a while. Per @lparrott  upon merge of this one we will be in a better state with widgets and the form rendering.

File: src/lwc/geFormService/geFormService.js [hunk 1]
```diff
@@ -83,12 +83,15 @@ class GeFormService {
...
     /**
-     * Takes a Data Import record, processes it, and returns the new Opportunity created from it.
+     * Takes a Data Import record and additional object data, processes it, and returns the new Opportunity created from it.
+     * @param createdDIRecord
+     * @param widgetValues
      * @returns {Promise<Id>}
      */
-    createOpportunityFromDataImport(createdDIRecord) {
+    createOpportunityFromDataImport(createdDIRecord, widgetValues) {
+        let widgetDataString = JSON.stringify(widgetValues);
```
Comment (bdvorachek) [inline src/lwc/geFormService/geFormService.js]: `const` instead of `let` here

File: src/classes/GE_FormRendererService.cls [hunk 2]
```diff
@@ -106,6 +111,45 @@ public with sharing class GE_FormRendererService {
 
     }
 
+    private static String convertWidgetDataToObjectJSON(String widgetData) {
+        // System.debug('widgetData: '); 
+        // System.debug(widgetData); 
+
+        List<SObject> thisList;
+        Map<String, Object> objNameToObjList = (Map<String, Object>)JSON.deserializeUntyped(widgetData);
```
Comment (bdvorachek) [inline src/classes/GE_FormRendererService.cls]: We need to handle the case where deserialization fails here, and check for null before attempting to deserialize.

**Comment by lparrott** [comment]
@bdvorachek @annejackson I made a few changes based on PR feedback, but didn't want to delay merging this since it's needed to continue Widget development, sorry for leaving so much work in progress!

**Comment by annejackson** [comment]
Dev review approved...merging to gift/entry branch.

**Pull request merged**